common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandResult.java

```diff
@@ -22,23 +22,45 @@
 //  THE SOFTWARE.
 package com.microsoft.identity.common.internal.controllers;
 
+import androidx.annotation.Nullable;
+
 public class CommandResult {
 
-    public enum ResultStatus { CANCEL, COMPLETED, ERROR };
+    public String getCorrelationId() {
+        return mCorrelationId;
+    }
+
+    public enum ResultStatus {
+        CANCEL,
+        COMPLETED,
+        ERROR;
+
+        public String getLogStatus() {
+            if (ResultStatus.COMPLETED == this) {
+                return "SUCCESS";
+            } else if (ResultStatus.ERROR == this) {
+                return "ERROR";
+            } else {
+                return "CANCEL";
+            }
+        }
+    }
 
     private ResultStatus mStatus;
     private Object mResult;
+    private String mCorrelationId;
 
-    public CommandResult(ResultStatus status, Object result){
+    public CommandResult(ResultStatus status, Object result, @Nullable String correlationId) {
```

Comment:
API breaking change here - can we make it additive?

Reply:
Yeah, we should add a new telescoping constructor.

Reply:
Yeah my bad, should've done that in the first place. will refactor it.

Reply:
LGTM